pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -656,16 +747,39 @@ func requestConfigByURL(getconfigCtx *getconfigContext, url string,
 		return invalidConfig, rv.TracedReqs
 	}
 
-	authWrappedRV := rv
-	decryptCtx := &cipher.DecryptCipherContext{
-		Log:                  log,
-		AgentName:            agentName,
-		PubSubControllerCert: getconfigCtx.pubControllerCert,
-		PubSubEdgeNodeCert:   getconfigCtx.zedagentCtx.subEdgeNodeCert,
+	var compoundConfig zconfig.CompoundEdgeDevConfig
+	if isCompoundConfig {
+		// Parse compound config and update the rv.RespContents with
+		// auth container stream for futher processing
+		err = parseCompoundDeviceConfig(getconfigCtx, &rv, &compoundConfig)
+		if err != nil {
+			log.Errorln("parseCompoundDeviceConfig: ", err)
+			// Inform ledmanager about cloud connectivity
+			utils.UpdateLedManagerConfig(log, types.LedBlinkConnectedToController)
+			getconfigCtx.ledBlinkCount = types.LedBlinkConnectedToController
+			publishZedAgentStatus(getconfigCtx)
+			return invalidConfig, rv.TracedReqs
+		}
 	}
-	err = zedcloud.RemoveAndVerifyAuthContainer(zedcloudCtx, decryptCtx, &rv, false)
+
+	// Copy of retval with auth container stream
+	var authWrappedRV zedcloud.SendRetval
+
+	// Decrypts auth container envelope if it is encrypted
+	err = decryptAuthContainer(getconfigCtx, &rv)
```

Comment:
This reads as if you (attempt to) decrypt even if !compoundConfig, which is confusing. Shouyld this be in the if statement above?

Reply:
I wanted to make decryption generic, regardless of the compound config, which is created only for the commands. But probably decryption won't be used for other cases. So makes sense to put it under the 'if' statement.